ui/apps/platform/cypress/helpers/request.js

```diff
@@ -96,22 +102,56 @@ export function interactAndWaitForResponses(
 }
 
 /**
- * Intercepts a GraphQL request during an interaction and yields the
- * variables object passed to the request's query
+ * Intercept requests and monitor requests/responses across multiple interactions
  *
- * @param {() => void} interactionCallback The interaction performed to trigger the GraphQL request
- * @param {string} opname The GraphQL operation name
- * @returns {Cypress.Chainable<Interception>}
+ * @template {string} RouteKey
+ * @param {Record<RouteKey, RouteMatcherOptions>} routeMatcherMap
+ * @param {Partial<Record<RouteKey, RouteHandler>>} [staticResponseMap]
+ * @returns {Promise<{
+ *    waitForRequests: typeof waitForRequests,
+ *    waitAndYieldRequestBodyVariables: typeof waitAndYieldRequestBodyVariables,
+ * }>} Helper functions used to monitor requests
+ *          after an interaction that causes a request
  */
-export function interactAndInspectGraphQLVariables(interactionCallback, opname) {
-    const url = `/api/graphql?opname=${opname}`;
+export function interceptAndWatchRequests(routeMatcherMap, staticResponseMap) {
+    interceptRequests(routeMatcherMap, staticResponseMap);
 
-    cy.intercept({ method: 'POST', url, times: 1 }).as(opname);
+    /**
+     * Wait for requests to complete after an interaction
+     *
+     * @param {RouteKey[]=} keys The keys of the routeMatcherMap to wait for, if not provided, wait for all keys in routeMatcherMap
+     * @param {WaitOptions=} waitOptions Wait options for cy.wait
+     * @returns {Cypress.Chainable<Interception> | Cypress.Chainable<Interception[]>} The interception object or array of interception objects
+     */
+    function waitForRequests(keys, waitOptions) {
+        const aliases =
+            keys && keys.length > 0
```

Comment:
Small question about style: pro and con to write truthy condition as `Array.isArray(keys)`

Reply:
I think this is subjective, but I prefer the `keys &&` idiom instead. To me this reads closer to `keys` being either nullish or an array, while `Array.isArray(keys)` reads closer to `keys` being either an array or **something else** (usually something else referring to a single item of the array's element type). This could just be due to how I've seen and used `isArray` in the product so far.